### ISSUE TITLE
Element with index attribute has broken attrs field when compiled

### ISSUE THREAD

**Issue by jpdutoit** [trueadm/babel-plugin-inferno#13]
```
<div index="text"></div>
```
compiles to:
```
({
	tag: "div",
	attrs: vtext,
	dom: null
});
```

**Comment by jpdutoit** [comment]
This was using the latest spike branch.

**Comment by JustinWilabyPrecor** [comment]
```js
<div index="text"></div>
```

`index="text"` is invalid html.  Did you mean `data-index="text"`?

**Comment by jpdutoit** [comment]
It also happens when passing a parameter called index to a component:
```
 <Component index="text"/>
```

**Comment by justinwilaby** [comment]
@jpdutoit - Can you confirm that #14 fixes the issue?

**Comment by jpdutoit** [comment]
Why keep the behaviour for number values? I couldn't see this documented
anywhere.

Will test it as soon as i get a chance.

**Comment by justinwilaby** [comment]
If you look at [how Inferno Works](https://github.com/trueadm/inferno/blob/master/docs/how-inferno-works.md), the `schema` is created with arguments whose name corresponds to the index of the template function prepended with `"v"` I don't think this is necessarily relevant for the spike branch since it seems there are changes that may obviate this but it makes sense in the `master` and `dev` branches of inferno. 

```js
<div selected={this.shouldBeSelected}></div>
```
Transpiles to:

```js
tpl3295379617 = Inferno.createTemplate(function (v0) {
    return {
        tag: 'div',
        attrs: {
            selected: v0
        }
    };
});

function render(){
    return tpl3295379617(this.shouldBeSelected);
}
```

Where `v0` represents the first argument in the template's `function`.

**Comment by justinwilaby** [comment]
I should add that adding the `index` on an attribute's schema descriptor is [here](https://github.com/trueadm/babel-plugin-inferno/blob/master/lib/index.js#L66)

**Comment by jpdutoit** [comment]
Yeah, createTemplate does not exist anymore in spike, so I thought perhaps
that part was a remnant of earlier code that has not been removed yet.
Unless it has some purpose...

**Comment by trueadm** [comment]
createTemplate and schemas have been removed in the `spike` branch. They are clunky in behaviour and made the codebase twice the size it should be with no gain in performance in most use-cases. I'm hoping to promote `spike` to `master` by Friday (in time for Easter) for both Inferno and this Babel plugin.

I've just got to sort out a few issues. If anyone is not on Slack, let me know so I can send you an invite. I'd love as much interaction before promoting to `master`, so we can squash the last few big bugs.
